default/python/AI/PolicyAI.py

```diff
@@ -395,6 +556,24 @@ def _deadopt(self, name: str) -> None:
                 self._ip += fo.getPolicy(name).adoptionCost()
             debug(f"Issued deadoption order for {name} turn {fo.currentTurn()}, remaining IP: {self._ip}")
 
+    def _deadopt_one_of(self, names: Set[str]) -> None:
+        """
+        Deadopt one of the given policies and store its name.
+        This can be used after a decision is made, or to temporarily deadopt one for
+        decision-making. In the later case, use _readopt_selected_one() to revert it.
+        """
+        selection = [(self._rate_policy(name), name) for name in names]
+        selection.sort()
+        debug(f"selection = {selection}")
+        for _, name in selection:
```

Comment:
Should we stop the cycle after finding one_of?

Reply:
Definitely, this is a real bug!